From b363bbdbedc0961e5fa0ee587f9f8619f882e7ae Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@cloudera.com>
Date: Tue, 6 Feb 2018 12:00:42 -0800
Subject: [PATCH 2735/2748] CLOUDERA-BUILD. CDH-62049. Short-circuit reads
 unnecessarily disabled for a long time.

Change-Id: Ib2e95ba7b160f69949d7daba1e9493f95e591ce3
(cherry picked from commit f128cf437cf5e88b2ae1ace31448bffc30b19454)
---
 .../java/org/apache/hadoop/hdfs/DFSConfigKeys.java |    2 +-
 .../src/main/resources/hdfs-default.xml            |    2 +-
 .../apache/hadoop/hdfs/TestBlockReaderFactory.java |    5 +++--
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSConfigKeys.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSConfigKeys.java
index ead420f..089f28e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSConfigKeys.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/DFSConfigKeys.java
@@ -710,7 +710,7 @@
   public static final String  DFS_DATANODE_SHARED_FILE_DESCRIPTOR_PATHS = "dfs.datanode.shared.file.descriptor.paths";
   public static final String  DFS_DATANODE_SHARED_FILE_DESCRIPTOR_PATHS_DEFAULT = "/dev/shm,/tmp";
   public static final String DFS_DOMAIN_SOCKET_DISABLE_INTERVAL_SECOND_KEY = "dfs.domain.socket.disable.interval.seconds";
-  public static final long DFS_DOMAIN_SOCKET_DISABLE_INTERVAL_SECOND_DEFAULT = 600;
+  public static final long DFS_DOMAIN_SOCKET_DISABLE_INTERVAL_SECOND_DEFAULT = 1;
   public static final String  DFS_SHORT_CIRCUIT_SHARED_MEMORY_WATCHER_INTERRUPT_CHECK_MS = "dfs.short.circuit.shared.memory.watcher.interrupt.check.ms";
   public static final int     DFS_SHORT_CIRCUIT_SHARED_MEMORY_WATCHER_INTERRUPT_CHECK_MS_DEFAULT = 60000;
   public static final String  DFS_NAMENODE_KEYTAB_FILE_KEY = "dfs.namenode.keytab.file";
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml b/hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml
index 37cbb6d..ca6e003 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml
@@ -2174,7 +2174,7 @@
 
 <property>
   <name>dfs.domain.socket.disable.interval.seconds</name>
-  <value>600</value>
+  <value>1</value>
   <description>
     The interval that a DataNode is disabled for future Short-Circuit Reads,
     after an error happens during a Short-Circuit Read. Setting this to 0 will
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestBlockReaderFactory.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestBlockReaderFactory.java
index ffaa5ee..7d2e0a1 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestBlockReaderFactory.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestBlockReaderFactory.java
@@ -315,8 +315,9 @@ public void run() {
    */
   @Test
   public void testShortCircuitCacheUnbufferDefault() throws Exception {
-    testShortCircuitCacheUnbufferWithDisableInterval(
-        DFS_DOMAIN_SOCKET_DISABLE_INTERVAL_SECOND_DEFAULT, true);
+    // test with upstream default interval, to make sure the disable works.
+    final long disableInterval = 600;
+    testShortCircuitCacheUnbufferWithDisableInterval(disableInterval, true);
   }
   /**
    * Test the case where if we disable the cache in
-- 
1.7.9.5

